partner_preview_scripts/ValidationLab.py

```diff
@@ -3,432 +3,472 @@
+                        f.col("id").alias("id"),
+                        f.col("tissueFromSource").alias("tissue"),
+                        f.col("tissueId"),
+                    )
+                ).alias("diseaseCellLines"),
+                "biomarkerList",
+                # Primary project:
+                "primaryProjectHit",
+                "primaryProjectId",
+            )
+        )
 
-    # The biomarker columns are unstacked into one single 'biomarkers' column:
-    biomarker_unstack = f"""stack({len(biomarkers_in_data)}, {", ".join([f"'{x}', {x}" for x in biomarkers_in_data])}) as (biomarker_name, biomarkers)"""
 
-    validation_lab_cell_lines = (
-        biomarkers
-        # Selecting cell line name, cell line annotation and applyting the stacking expression:
-        .select(
-            f.col("cellName"),
-            "diseaseCellLines",
-            f.expr(biomarker_unstack),
+@dataclass
+class ValidationLabProjectParser:
+    diseaseFromSourceMappedId: str
+    diseaseFromSource: str
+    tissueId: str
+    experimentDataFile: str
+    biomarkerDataFile: str
+    studyOverview: str
+    primaryProjectId: str
+    releaseVersion: str
+    releaseDate: str
+    excludeStudy: bool
+
+    def parse_evidence(
+        self: ValidationLabProjectParser,
+        spark: SparkSession,
+        input_path: str,
+        assays: list,
+    ) -> DataFrame:
+        logger = logging.getLogger(__name__)
+
+        validation_input_file = f"{input_path}/{self.experimentDataFile}"
+        biomarker_file = f"{input_path}/{self.biomarkerDataFile}"
+
+        logger.info(f"Reading project data: {validation_input_file}")
+        logger.info(f"Reading biomarker data: {biomarker_file}")
+
+        # Reading data:
+        raw_experiment_data = spark.read.option("multiline", True).csv(
+            validation_input_file, sep="\t", header=True
         )
-        # Filter out all null biomarkers:
-        .filter(
-            (f.col("biomarkers").isNotNull())
-            &
-            # Following the request of the validation lab, we are removing CRIS biomarker annotation:
-            (f.col("biomarker_name") != "CRIS_subtype")
+        # First round of processing:
+        processed_experiment = self._process_raw_evidence(raw_experiment_data, assays)
+
+        # Reading biomarker data:
+        raw_biomarkers = spark.read.csv(biomarker_file, sep="\t", header=True)
+
+        # Formatting assays (depends on the formatted experiment data):
+        parsed_assays = self._parser_assay_object(processed_experiment, assays, spark)
+
+        # Get parsed biomarkers:
+        biomarkers = BiomarkerParser.get_biomarkers(raw_biomarkers)
+
+        # attributes added as column:
+        columns_to_add = [
+            "diseaseFromSourceMappedId",
+            "diseaseFromSource",
+            "tissueId",
+            "studyOverview",
+            "primaryProjectId",
+            "releaseVersion",
+            "releaseDate",
```

Comment:
The config says the data is from 2022, is this date up to date?

Reply:
Oh, I forget to do the update.

Reply:
Also updated the ppp config files under `gs://otar013-ppp/PPP-evidence-configuration/`